core/modules/indexers/backlinks-index.js

```diff
@@ -0,0 +1,108 @@
+/*\
+title: $:/core/modules/indexers/backlinks-indexer.js
+type: application/javascript
+module-type: indexer
+
+Indexes the tiddlers' backlinks
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global modules: false */
+"use strict";
+
+
+function BacklinksIndexer(wiki) {
+	this.wiki = wiki;
+}
+
+BacklinksIndexer.prototype.init = function() {
+	this.index = null;
+}
+
+BacklinksIndexer.prototype.rebuild = function() {
+	this.index = null;
+}
+
+BacklinksIndexer.prototype._getLinks = function(tiddler) {
+	var links = [],
+            parser =  this.wiki.parseText(tiddler.fields.type, tiddler.fields.text, {}),
+	    checkParseTree = function(parseTree) {
```

Comment:
I'm not super thrilled about the duplication betwen here and `getTiddlerLinks` in wiki.js.  I wrote this a while ago and can't exactly remember why I didn't just use `getTiddlerLinks` - maybe I wanted to avoid the caching layer?

Reply:
Ah, I see why now - I needed to be able to parse an arbitrary tiddler object that I pass in, rather than get from the tiddler store, because the `update` method gets such a tiddler object via the update descriptor.

Reply:
your .init does nothing, but it could use the existing core cache, if there is one.

Reply:
Your indexer doesn't update the tiddler cache. .. So work has to be done twice. .. I also don't understand why you don't use: `wiki.getTiddlerLinks` .. IMO it does the same thing and it also updated the tiddler cache. 

You are duplicating code here, which makes maintenance problematic. If we update `wiki.getTiddlerLinks` we will forget about the indexer and vice versa. 

So if you need new functionality from `wiki.getTiddlerLinks` you need to update it there. ...

Just my opinion.

Reply:
>Ah, I see why now - I needed to be able to parse an arbitrary tiddler object that I pass in, rather than get from the tiddler store, because the update method gets such a tiddler object via the update descriptor.

The `addTiddler` function saves the tiddler at: https://github.com/Jermolene/TiddlyWiki5/blob/b258afea0dc564b737409d88c72ca4180f967799/boot/boot.js#L1077

The indexers are called at: https://github.com/Jermolene/TiddlyWiki5/blob/b258afea0dc564b737409d88c72ca4180f967799/boot/boot.js#L1092

So it's perfectly fine, that you use the `wiki` functions. .. IMO no code duplication needed.

Reply:
@pmario I agree with you on the problem on code duplication and would prefer to leverage `getTiddlerLinks` if I can.  The problem is how can I call `getTiddlerLinks` and get the links for the _former_ version of the tiddler (I need to remove any links that have disappeared in the new version), since `getTiddlerLinks` takes a title and uses that to get the tiddler from the tiddler store.

Reply:
FWIW, I have no objections to updating the cache - I don't remember exactly why I omitted it from my reimplementation.  Maybe because it felt like storing too much data for something that might not be used again?

Reply:
>...Maybe because it felt like storing too much data for something that might not be used again?

The cache is also used by Orphan and Missing tiddlers. So it can be reused. ... As you can see by my later replies, I did see the problem.  That's why, I did ask Jeremy.

Reply:
Yeah, that's ok with me!

Reply:
I've extracted that code into its own method and changed the indexer to use it.